population/create_glpi_computer_redfish.py

```diff
@@ -787,6 +797,100 @@ def post_to_glpi(  # noqa: C901
+        plugin_response (list): list of requests objects returned by BMC API endpoint
+        computer_id (int): ID of the computer associated with the BMC Address
+        redfish_base_url (str): URL used to connect to Redfish
+        comment (str): Comment of computer in GLPI. None if this field is empty
+        put (bool): If this computer already exists in GLPI
+
+    Returns:
+        glpi_post (dict): Contains information about a Computer to be passed to GLPI
+    """
+    # plugin_response: list of response objects. meant to be from check_fields.
+    if "ERROR" in plugin_response[0].json()[0]:
+        glpi_post = add_bmc_address_to_comments(glpi_post, comment, redfish_base_url)
+        print(
+            "The provided field endpoint is unavailable, "
+            + "adding the BMC address to the comments."
+        )
+    else:
+        glpi_post = set_bmc_address_field(
+            glpi_post, plugin_response, computer_id, redfish_base_url, put
+        )
+
+    return glpi_post
+
+
+def add_bmc_address_to_comments(
+    glpi_post: dict, redfish_base_url: str, comment: str
+) -> dict:
+    """Add BMC address to comments of GLPI post.
+
+    Args:
+        glpi_post (dict): Contains information about a Computer to be passed to GLPI
+        redfish_base_url (str): URL used to connect to Redfish
+        comment (str): Comment of computer in GLPI. None if this field is empty
+
+    Returns:
+        glpi_post (str): Contains information about a Computer to be passed to GLPI
+    """
+    if comment:
+        if "BMC Address" not in comment:
+            glpi_post["comment"] = (
+                comment + "\nBMC Address: " + redfish_base_url.partition("https://")[2]
+            )
+    else:
+        glpi_post["comment"] = (
+            "BMC Address: " + redfish_base_url.partition("https://")[2]
+        )
+
+    return glpi_post
+
+
+def set_bmc_address_field(
+    glpi_post: dict,
+    plugin_response: list,
+    computer_id: int,
+    redfish_base_url: str,
+    put: bool,
+) -> dict:
+    """Set BMC address field of GLPI post if it's empty.
+
+    Args:
+        glpi_post (dict): Contains information about a Computer to be passed to GLPI
+        plugin_response (list): list of requests objects returned by  BMC API endpoint
+        computer_id (int): ID of the computer associated with the BMC Address
+        redfish_base_url (str): URL used to connect to Redfish
+        put (bool): If this computer already exists in GLPI
+
+    Returns:
+        glpi_post (dict): Contains information about a Computer to be passed to GLPI
+    """
+    if put:
+        for computer_group in plugin_response:
+            for computer in computer_group.json():
+                if computer["items_id"] == computer_id:
+                    if computer["bmcaddressfield"]:
+                        return glpi_post
+        glpi_post["bmcaddressfield"] = redfish_base_url.partition("https://")[2]
```

Comment:
Can't you just put this command outside the if statement, then it will happen on the if or the else without explicitly doing that?

Reply:
Good catch - fixed in latest commit